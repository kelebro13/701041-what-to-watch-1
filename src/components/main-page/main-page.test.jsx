import renderer from "react-test-renderer";
import {MemoryRouter} from 'react-router-dom';
import MainPage from "./main-page";

it(`render correctly App component`, () => {
  const genre = `All genres`;
  const genres = [`All genres`, `Kids & Family`, `Comedies`];
  const films = [
    {
      name: `We need to talk about Kevin`,
      posterImage: `https://es31-server.appspot.com/wtw/static/film/poster/We_need_to_talk_about_Kevin.jpg`,
      previewImage: `https://es31-server.appspot.com/wtw/static/film/preview/we-need-to-talk-about-kevin.jpg`,
      backgroundImage: `https://es31-server.appspot.com/wtw/static/film/background/We_need_to_talk_about_Kevin.jpg`,
      backgroundColor: `#E1DFDE`,
      description: `Kevin's mother struggles to love her strange child, despite the increasingly dangerous things he says and does as he grows up. But Kevin is just getting started, and his final act will be beyond anything anyone imagined.`,
      rating: 7.5,
      scoresCount: 123240,
      director: `Lynne Ramsay`,
      starring: [
        `Tilda Swinton`,
        `John C. Reilly`,
        `Ezra Miller`
      ],
      runTime: 112,
      genre: `Drama`,
      released: 2011,
      id: 2,
      isFavorite: false,
      videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
      previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
    },
    {
      name: `The Revenant`,
      posterImage: `https://es31-server.appspot.com/wtw/static/film/poster/Revenant.jpg`,
      previewImage: `https://es31-server.appspot.com/wtw/static/film/preview/revenant.jpg`,
      backgroundImage: `https://es31-server.appspot.com/wtw/static/film/background/Revenant.jpg`,
      backgroundColor: `#92918B`,
      description: `A frontiersman on a fur trading expedition in the 1820s fights for survival after being mauled by a bear and left for dead by members of his own hunting team.`,
      rating: 8,
      scoresCount: 618498,
      director: `Alejandro G. Iñárritu`,
      starring: [
        `Leonardo DiCaprio`,
        `Tom Hardy`,
        `Will Poulter`
      ],
      runTime: 156,
      genre: `Action`,
      released: 2015,
      id: 3,
      isFavorite: false,
      videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
      previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
    },
    {
      name: `Bronson`,
      posterImage: `https://es31-server.appspot.com/wtw/static/film/poster/bronson.jpg`,
      previewImage: `https://es31-server.appspot.com/wtw/static/film/preview/bronson.jpg`,
      backgroundImage: `https://es31-server.appspot.com/wtw/static/film/background/bronson.jpg`,
      backgroundColor: `#73B39A`,
      description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
      rating: 7.1,
      scoresCount: 109661,
      director: `Nicolas Winding Refn`,
      starring: [
        `Tom Hardy`,
        `Kelly Adams`,
        `Luing Andrews`
      ],
      runTime: 92,
      genre: `Action`,
      released: 2008,
      id: 4,
      isFavorite: false,
      videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
      previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
    },
  ];

  const tree = renderer
    .create(
        <MemoryRouter>
          <MainPage genre={genre} genres={genres} filmsByGenre={films} film={films[0]} changeSelectedGenre={() => {}} setFilmsByGenre={() => {}}/>
        </MemoryRouter>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
