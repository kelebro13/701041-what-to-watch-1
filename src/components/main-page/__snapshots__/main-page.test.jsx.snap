// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly App component 1`] = `
Array [
  <section
    className="movie-card"
  >
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          href="/mylist"
          onClick={[Function]}
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="Oliver.conner"
              height="63"
              src="https://es31-server.appspot.com//img/1.png"
              width="63"
            />
          </div>
        </a>
      </div>
    </header>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            onClick={[Function]}
          >
            Kids & Family
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            onClick={[Function]}
          >
            Comedies
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="175"
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/we-need-to-talk-about-kevin.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              We need to talk about Kevin
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="175"
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/revenant.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              The Revenant
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="175"
              muted={true}
              poster="https://es31-server.appspot.com/wtw/static/film/preview/bronson.jpg"
              src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Bronson
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
